src/GovUk.Education.ExploreEducationStatistics.Common/ViewModels/ErrorViewModel.cs

```diff
@@ -32,4 +34,17 @@ public record ErrorViewModel
     /// more context to users. May be omitted if there is none.
     /// </summary>
     public object? Detail { get; init; }
+
+    public static ErrorViewModel Create(ValidationFailure failure)
+    {
+        var detail = failure.GetErrorDetail();
+
+        return new ErrorViewModel
+        {
+            Path = failure.PropertyName,
+            Code = failure.ErrorCode.Replace("Validator", ""),
+            Message = failure.ErrorMessage,
+            Detail = detail.Count > 0 ? detail : null
+        };
```

Comment:
Cool, I think this fixes the problem I was having with getting the additional details from the failure, and the reason why I had to put a custom message into my rule to identity the name of the field! Thanks.